app/src/main/java/com/quarkworks/apartmentgroceries/grocery/GroceryCell.java

```diff
@@ -45,12 +55,36 @@ private void initialize() {
         nameTextView = (TextView) findViewById(R.id.grocery_cell_grocery_name_id);
         createdByTextView = (TextView) findViewById(R.id.grocery_cell_created_by_id);
         purchasedByTextView = (TextView) findViewById(R.id.grocery_cell_purchased_by_id);
+        statusTextView = (TextView) findViewById(R.id.grocery_cell_grocery_status_id);
+        photoImageView = (ImageView) findViewById(R.id.grocery_cell_created_by_image_view_id);
     }
 
     public void setViewData(final RGroceryItem groceryItem, final int position){
         nameTextView.setText(groceryItem.getName());
-        createdByTextView.setText(groceryItem.getCreatedBy());
-        purchasedByTextView.setText(groceryItem.getPurchasedBy());
+
+        String purchasedBy = groceryItem.getPurchasedBy();
+        if (TextUtils.isEmpty(purchasedBy)) {
+            statusTextView.setText(getResources().getString(R.string.grocery_cell_item_status_open));
+        } else {
+            purchasedByTextView.setText("by " + purchasedBy);
+            statusTextView.setText(getResources().getString(R.string.grocery_cell_item_status_closed));
+        }
+        statusTextView.setTextColor(getResources().getColor(R.color.colorPrimary));
+
+        photoImageView.setImageResource(R.drawable.ic_launcher);
+        SyncUser.getById(groceryItem.getCreatedBy()).continueWith(new Continuation<RUser, Void>() {
```

Comment:
Right now, your code makes a network request every time the view data gets set, which is not a best practice. I will let you figure out a better way to do this.